packages/mobile-top-nav/src/media-slider.js

```diff
@@ -1,14 +1,73 @@
-import { LitElement, html, css } from 'lit-element';
+import { LitElement, html } from 'lit-element';
+import mediaSliderCss from './css/media-slider';
+import menus from './data/menus';
+import './wayback-search';
+import './more-slider';
 
 class MediaSlider extends LitElement {
+  constructor() {
+    super();
+    const defaultLinks = { iconLinks: [], featuredLinks: [], links: [] };
+
+    this.mediaSliderOpen = false;
+    this.mediaSliderAnimate = false;
+    this.selectedMenuOption = 'texts';
+
+    // Begin properties not monitored by LitElement
+    this.links = menus[this.selectedMenuOption] || defaultLinks;
```

Comment:
I would make `links` a reactive property since there are several dependencies down the line like in `renderLinks` and `renderIconLinks`
```
get links() {
  return menus[this.selectedMenuOption] || this.defaultLinks;
}
```
I think this would let you get rid of that `shouldUpdate` lifecycle event below.

Reply:
The `shouldUpdate` behaves slightly different than an either/or. When the subnav is closed, if we use a getter like above, the menu is replaced with an empty one and it's visible as it animates out. Instead, I'm leaving the current menu in place when `selectedMenuOption` is unset.

Reply:
Oh I see.. hmm since that has dependent logic on the changing of `selectedMenuOption`, it might be a good candidate for the `updated(changedProperties)` callback instead of `shouldUpdate`. `updated()` is the more common place to handle property-dependent changes. I would probably still make `links` a property, though so any listeners can update when it changes. 

ie:

```
updated(changedProperties) {
  if (changedProperties.has('selectedMenuOption')) {
    this.handleSelectedMenuOptionChange();
  }
}

handleSelectedMenuOptionChange() {
  if (menus[this.selectedMenuOption]) {
    this.links = menus[this.selectedMenuOption];
  }
}
```